gcm_filters/kernels.py

```diff
@@ -51,6 +58,95 @@ def __call__(self, field: ArrayType):
 ALL_KERNELS[GridType.CARTESIAN] = CartesianLaplacian
 
 
+@dataclass
+class MOM5LaplacianU(BaseLaplacian):
+    """Laplacian for MOM5 gird (velocity points)"""
+
+    dxt: ArrayType
+    dyt: ArrayType
+    dxu: ArrayType
+    dyu: ArrayType
+    area_u: ArrayType
+    wet: ArrayType
+
+    def __post_init__(self):
+        np = get_array_module(self.wet)
+
+        self.x_wet_mask = self.wet * np.roll(self.wet, -1, axis=-1)
+        self.y_wet_mask = self.wet * np.roll(self.wet, -1, axis=-2)
+
+    def __call__(self, field: ArrayType):
+        """Uses code by Elizabeth"""
+        np = get_array_module()
+        field = np.nan_to_num(field)
+        fx = 2 * (np.roll(field, shift=-1, axis=-2) - field)
+        fx /= np.roll(self.dxt, -1, axis=-2) + np.roll(self.dxt, (-1, -1), axis=(0, 1))
```

Comment:
If we are striving for optimal performance in the long term, I would move calculations like this one to `__post_init__` in order to not repeat these calculations for every depth level and time step. (But I'm sure I start sounding like a broken record, since I have been saying this pretty much in every PR so far. :smile: )